File: app/src/test/java/com/example/megaport/mynews/NotificationsPreferencesUnitTest.java
package com.example.megaport.mynews;


import com.example.megaport.mynews.Models.notifications.NotificationsPreferences;

import org.junit.Test;
import java.util.ArrayList;
import java.util.List;

import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertFalse;
import static junit.framework.Assert.assertTrue;

public class NotificationsPreferencesUnitTest {

    @Test
    public void createNotificationsPreferences() {
        List<String> categoryList = new ArrayList<>();
        String queryTerm = "test";

        NotificationsPreferences notificationsPref = new NotificationsPreferences(queryTerm, categoryList, false);

        assertEquals("test", notificationsPref.getQueryTerm());
        assertTrue( notificationsPref.getCategoryList().isEmpty() );
        assertFalse( notificationsPref.isEnabled() );
    }

    @Test
    public void setQueryTerm() {
        NotificationsPreferences notificationsPref = new NotificationsPreferences("",null, false);

        notificationsPref.setQueryTerm("test");

        assertEquals("test", notificationsPref.getQueryTerm());
    }

    @Test
    public void setCategoryList() {
        NotificationsPreferences notificationsPref = new NotificationsPreferences("",null, false);

        List<String> categoryList = new ArrayList<>();
        String entry1 = "Arts";
        String entry2 = "Politics";
        categoryList.add(entry1);
        categoryList.add(entry2);

        notificationsPref.setCategoryList(categoryList);

        assertEquals("Politics", notificationsPref.getCategoryList().get(1));
    }

    @Test
    public void setEnabled() {
        NotificationsPreferences notificationsPref = new NotificationsPreferences("",null, false);

        notificationsPref.setEnabled(true);

        assertTrue( notificationsPref.isEnabled() );
    }
}

File: app/src/test/java/com/example/megaport/mynews/NotificationsPreferencesUnitTest.kt
package com.example.megaport.mynews

import com.example.megaport.mynews.models.notifications.NotificationsPreferences
import org.junit.Test
import java.util.ArrayList
import junit.framework.Assert.assertEquals
import junit.framework.Assert.assertFalse
import junit.framework.Assert.assertTrue

class NotificationsPreferencesUnitTest {

    @Test
    fun createNotificationsPreferences() {
        val categoryList = ArrayList<String>()
        val queryTerm = "test"

        val notificationsPref = NotificationsPreferences(queryTerm, categoryList, false)

        assertEquals("test", notificationsPref.queryTerm)
        assertTrue(notificationsPref.categoryList!!.isEmpty())
        assertFalse(notificationsPref.isEnabled)
    }

    @Test
    fun setQueryTerm() {
        val notificationsPref = NotificationsPreferences("", null, false)

        notificationsPref.queryTerm = "test"

        assertEquals("test", notificationsPref.queryTerm)
    }

    @Test
    fun setCategoryList() {
        val notificationsPref = NotificationsPreferences("", null, false)

        val categoryList = ArrayList<String>()
        val entry1 = "Arts"
        val entry2 = "Politics"
        categoryList.add(entry1)
        categoryList.add(entry2)

        notificationsPref.categoryList = categoryList

        assertEquals("Politics", notificationsPref.categoryList!![1])
    }

    @Test
    fun setEnabled() {
        val notificationsPref = NotificationsPreferences("", null, false)

        notificationsPref.isEnabled = true

        assertTrue(notificationsPref.isEnabled)
    }
}